quolance-ui/src/app/(with-main-layout)/(authenticated-pages)/profile/components/update-password-form.tsx

```diff
@@ -0,0 +1,119 @@
+"use client";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import React from "react";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { z } from "zod";
+
+import { useAuthGuard } from "@/lib/auth/use-auth";
+import httpClient from "@/lib/httpClient";
+
+import ErrorFeedback from "@/components/error-feedback";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+
+import { HttpErrorResponse } from "@/models/http/HttpErrorResponse";
+
+const schema = z
+  .object({
+    oldPassword: z.string().optional(),
+    password: z.string().min(8),
+    confirmPassword: z.string().min(8),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    message: "Passwords do not match",
+    path: ["confirmPassword"],
+  });
+
+type Schema = z.infer<typeof schema>;
+export default function UpdatePasswordForm() {
+  const { user, mutate } = useAuthGuard({ middleware: "auth" });
+  const [errors, setErrors] = React.useState<HttpErrorResponse | undefined>(
+    undefined
+  );
+
+  const onSubmit = (data: Schema) => {
```

Comment:
Since we are using reactQuery, we might want to use  [useMutation](https://tanstack.com/query/v4/docs/framework/react/reference/useMutation), which can be implemented in [userHooks.tsx](https://github.com/abdelh17/Quolance/blob/main/quolance-ui/src/hooks/userHooks.tsx)

Reply:
@FehamIsmail @ChemsCode this comment is a blocker on the PR, resolve if not needed.

Reply:
@FehamIsmail @abdelh17 We'll implement this use of ReactQuery in anohter iteration.